### PR TITLE
[NTOSKRNL] Implement JobObjectBasicProcessIdList

### PR DESCRIPTION
**Pull request by learn-more**

### CODE REVIEW

File: ntoskrnl/ps/job.c [hunk 1]
```diff
@@ -707,6 +708,60 @@ NtQueryInformationJobObject (
...
+                    Status = STATUS_SUCCESS;
+
+                    for (NextEntry = Job->ProcessListHead.Flink;
+                        NextEntry != &Job->ProcessListHead;
+                        NextEntry = NextEntry->Flink)
+                    {
+                        if (Total == Job->ActiveProcesses)
+                        {
+                            DPRINT1("ERROR: Job->ActiveProcesses did not match Job->ProcessListHead!");
+                            Status = STATUS_PROCESS_IN_JOB;
```
Comment (learn-more) [inline ntoskrnl/ps/job.c:741]: Not sure what to return here.

**Review by oleg-dubinskiy — COMMENTED** [review]
Suggestion for better commit message:
`[NTOS:PS] Implement JobObjectBasicProcessIdList `

File: ntoskrnl/ps/job.c (hunk 1 above)
Comment (GeoB99) [reply ntoskrnl/ps/job.c:741]: I do not think you're actually supposed to return a status code in this kind of situation. Whenever a process' unique ID is not associated within a job, `NumberOfAssignedProcesses` field is decremented and the operation moves on until you have looped thorough all the present processes in the job.

With that said, `NumberOfAssignedProcesses` has to be set to active processes and when a process' ID is indeed associated with that job, you increment `NumberOfProcessIdsInList` and submit the process ID to the list.

Comment (learn-more) [reply ntoskrnl/ps/job.c:741]: This `ActiveProcesses` count is relied upon earlier to ensure the buffer was big enough.
If there is a mismatch between this number and the LIST_ENTRY there is probably a serious problem?
Might even warrant an assert.

**Pull request closed without merging**